Makefile

```diff
@@ -4,16 +4,31 @@ export GOOS=$(shell go env GOOS)
 export GOARCH=$(shell go env GOARCH)
 export GOVERSION=$(shell go list -m -f '{{.GoVersion}}')
 
-ifeq ($(OS), Windows_NT)
+ifndef VERSION
+ifndef SNAPSHOT_VERSION
+ifeq ($(GOOS), windows)
 	VERSION := $(shell git describe --exact-match --tags 2>nil)
 else
 	VERSION := $(shell git describe --exact-match --tags 2>/dev/null)
 endif
+endif
+endif
 COMMIT := $(shell git rev-parse --short HEAD)
 
 LDFLAGS := $(LDFLAGS) -X main.commit=$(COMMIT) -X main.date=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 ifdef VERSION
 	LDFLAGS += -X main.version=$(VERSION)
+	GORELEASER_SETENV := GORELEASER_CURRENT_TAG=$(VERSION)
+else
+ifndef SNAPSHOT_VERSION
+	SNAPSHOT_VERSION := dev
+endif
+ifndef SNAPSHOT_PACKAGE_TAG
+	SNAPSHOT_PACKAGE_TAG := dev
+endif
+	LDFLAGS += -X main.version=$(SNAPSHOT_VERSION)
+	GORELEASER_SETENV := SNAPSHOT_VERSION=$(SNAPSHOT_VERSION) SNAPSHOT_PACKAGE_TAG=$(SNAPSHOT_PACKAGE_TAG)
+	SNAPSHOT_FLAG := --snapshot
```

Comment:
Irritating, it seems `--snapshot` disables uploading to S3, so we can't use it for nightlies... investigating alternatives